src/model/underlying_mdp.jl

```diff
@@ -0,0 +1,33 @@
+# provide a structure to extract the underlying MDP of a POMDP
+
+struct UnderlyingMDP{P, S, A} <: MDP{S, A} where P <: POMDP{S, A, Any}

```

Comment:
I don't think this is working like you think it is (In fact, I'm not sure how it is even running). TigerPOMDP, for instance, is not a subtype of POMDP{S, A, Any}. See the warning in this section https://docs.julialang.org/en/stable/manual/types/#Parametric-Composite-Types-1

I think the `where` might just be getting ignored because I can do this:
```
julia> struct A{T} <: MDP{Int, Int} where T <: Number
       t::T
       end

julia> A("test")
A{String}("test")
```

I think it is sufficient to do
```
struct UnderlyingMDP{P<:POMDP, S, A} <: MDP{S, A}
```
and just enforce the S, A constraints in the constructor